codegen/java-gen/src/main/scala/com/lightbend/akkasls/codegen/java/ViewServiceSourceGenerator.scala

```diff
@@ -53,146 +57,226 @@ object ViewServiceSourceGenerator {
+        "com.google.protobuf.EmptyProto",
+        "java.util.function.Function"
+      )
+    )
+
+    s"""|$managedCodeCommentString
+        |package $packageName;
+        |
+        |$imports
+        |
+        |public class ${view.providerName} implements ViewProvider {
+        |
+        |  private final Function<ViewCreationContext, ${view.className}> viewFactory;
+        |
+        |  /** Factory method of ${view.className} */
+        |  public static ${view.providerName} of(
+        |      Function<ViewCreationContext, ${view.className}> viewFactory) {
+        |    return new ${view.providerName}(viewFactory);
+        |  }
+        |
+        |  private ${view.providerName}(
+        |      Function<ViewCreationContext, ${view.className}> viewFactory) {
+        |    this.viewFactory = viewFactory;
+        |  }
+        |
+        |  @Override
+        |  public String viewId() {
+        |    return "${view.viewId}";
+        |  }
+        |
+        |  @Override
+        |  public final Descriptors.ServiceDescriptor serviceDescriptor() {
+        |    return ${view.fqn.parent.javaOuterClassname}.getDescriptor().findServiceByName("${view.fqn.name}");
+        |  }
+        |
+        |  @Override
+        |  public final ${view.handlerName} newHandler(ViewCreationContext context) {
+        |    return new ${view.handlerName}(viewFactory.apply(context));
+        |  }
+        |
+        |  @Override
+        |  public final Descriptors.FileDescriptor[] additionalDescriptors() {
+        |    return new Descriptors.FileDescriptor[] {${view.fqn.parent.javaOuterClassname}.getDescriptor()};
+        |  }
+        |}""".stripMargin
+  }
+
+  private[codegen] def viewSource(
+      view: ModelBuilder.ViewService,
+      packageName: String
+  ): String = {
+
+    val imports = generateImports(
+      view.commandTypes,
+      packageName,
+      Seq(
+        "com.akkaserverless.javasdk.view.View",
+        "com.akkaserverless.javasdk.view.ViewContext",
+        "java.util.function.Function"
+      )
+    )
+
+    val handlers = view.transformedUpdates.map { update =>
+      val stateType = qualifiedType(update.outputType)
+      s"""@Override
+         |public UpdateEffect<${stateType}> ${lowerFirst(update.fqn.name)}(
+         |  $stateType state, ${qualifiedType(update.inputType)} ${lowerFirst(update.fqn.name)}) {
+         |  throw new RuntimeException("Update handler for '${update.fqn.name}' not implemented yet");
+         |}""".stripMargin
+    }
+
+    s"""$managedCodeCommentString
+       |package $packageName;
+       |
+       |$imports
+       |
+       |public class ${view.className} extends ${view.interfaceName} {
+       |
+       |  public ${view.className}(ViewContext context) {}
+       |
+       |  @Override
+       |  public ${qualifiedType(view.state.fqn)} emptyState() {
+       |    throw new RuntimeException("Empty state for '${view.className}' not implemented yet");
```

Comment:
in the entities we have:
```
throw new UnsupportedOperationException("Not implemented yet, replace with your empty entity state");
```

Reply:
Regarding generating a default empty state or not. I think this is fine, also for transform_updates=false because it will not be called in that case.

Reply:
Was just arriving at the same thought.